### PR TITLE
Add ansible playbook equivalent of fsdp_setup

### PR DESCRIPTION
**Pull request by CodyChengIOL**
This is the Ansible version of the fsdp_setup shell scripts and so each task file is named directly after each function in the rdma_functions.sh script. 

The setup_host.yml playbook is the main file that is used to run every tasks necessary to setup the nodes. The setup_host.yml playbook runs the roles common, interfaces, dhcp, and nfs in the order that was listed. Each of these roles have a main.yml which runs the tasks that relate to that role's purpose. The common role runs tasks that edits configuration files, setup ssh, etc. The interfaces role creates the interfaces, and creates the Network Manager dispatcher scripts that go along with the interfaces. The dhcp role creates the dhcp client files and uploads them to builder-00. 

I have left comments throughout some of the files about any questions I have.

Signed-off-by: Cody Cheng <ccheng@iol.unh.edu>

### CODE REVIEW

**Comment by CodyChengIOL** [comment]
Note: Currently, I am testing using this Ansible playbook to setup the nodes and I am running to an issue where it seems like the network interfaces are being created in the Anaconda environment but then when it exits that Anaconda environment, they revert back to the original interfaces.

**Comment by dledford** [comment]
Your second patch, labeled Cleanup, is missing the Signed-off-by: portion of the commit message and can't be merged without it.  You can force push an update to the branch with the sign off added to that commit.

**Comment by CodyChengIOL** [comment]
Update: This Ansible playbook works now by using beaker snippets to setup a systemd service which will run the Ansible playbook on boot after the installation and setup.

**Comment by CodyChengIOL** [comment]
This is now able to run the Ansible Playbook during the Beaker post installation stage using the system_post kickstart in this PR: https://github.com/OpenFabrics/fsdp_beaker_snippets/pull/27

File: ansible/group_vars/all.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+lab_networks: 10.12.0.0/16
+ib0_2k_nets:
+  - ib0
+  - ib0.2
+ib0_4k_nets:
+  - ib0.4
+  - ib0.6
+ib0_nets: "{{ ib0_2k_nets + ib0_4k_nets }}"
+
+# InfiniBand Fabric 2
```
Comment (PatrickRobbIOL) [inline ansible/group_vars/all.yml:15]: Should these commented out configs stay?

Comment (CodyChengIOL) [reply ansible/group_vars/all.yml:15]: I'm not sure. Looking at the git history of rdma-functions.sh, it seems like these commented out configs have never been used but maybe they could be useful in the future.

**Pull request merged**